src/module/canvas/token/object.ts

```diff
@@ -309,7 +309,7 @@ class TokenPF2e extends Token<TokenDocumentPF2e> {
             const { width, height, texture } = source;
             this.document.reset();
             // If any of the following changed, a full redraw should be performed
-            const postChange = pick(this.document, ["width", "height", "texture"]);
+            const postChange = pick(this.document.toObject(), ["width", "height", "texture"]);
```

Comment:
The current code here has a bug, but prepared data is what needs to be compared against.

Reply:
I was just happy I found the root cause and this seemed to fix it. 😄